spotinst/resource_spotinst_ocean_extended_resourcce_definition.go

```diff
@@ -0,0 +1,198 @@
+
+	input := &aws.ReadExtendedResourceDefinitionInput{ExtendedResourceDefinitionID: spotinst.String(resourceId)}
+	resp, err := meta.(*Client).ocean.CloudProviderAWS().ReadExtendedResourceDefinition(context.Background(), input)
+	if err != nil {
+		// If the ExtendedResourceDefinition was not found, return nil so that we can show
+		// that the ExtendedResourceDefinition does not exist
+		if errs, ok := err.(client.Errors); ok && len(errs) > 0 {
+			for _, err := range errs {
+				if err.Code == ErrCodeExtendedResourceDefinitionNotFound {
+					resourceData.SetId("")
+					return nil
+				}
+			}
+		}
+		return fmt.Errorf("failed to read extended resource definition: %s", err)
+	}
+
+	// If nothing was found, then return no state.
+	ExtendedResourceDefinitionResponse := resp.ExtendedResourceDefinition
+	if ExtendedResourceDefinitionResponse == nil {
+		resourceData.SetId("")
+		return nil
+	}
+
+	if err := commons.OceanAWSExtendedResourceDefinitionResource.OnRead(ExtendedResourceDefinitionResponse, resourceData, meta); err != nil {
+		return err
+	}
+	log.Printf("===> ExtendedResourceDefinition read successfully: %s <===", resourceId)
+	return nil
+}
+
+func resourceSpotinstOceanAWSExtendedResourceDefinitionCreate(resourceData *schema.ResourceData, meta interface{}) error {
+
+	log.Printf(string(commons.ResourceOnCreate), commons.OceanAWSExtendedResourceDefinitionResource.GetName())
+
+	extendedResourceDefinition, err := commons.OceanAWSExtendedResourceDefinitionResource.OnCreate(resourceData, meta)
+	if err != nil {
+		return err
+	}
+	extendedResourceDefinitionId, err := createOceanAWSExtendedResourceDefinition(resourceData, extendedResourceDefinition, meta.(*Client))
+	if err != nil {
+		return err
+	}
+
+	resourceData.SetId(spotinst.StringValue(extendedResourceDefinitionId))
+
+	log.Printf("===> ExtendedResourceDefinition created successfully: %s <===", resourceData.Id())
+
+	return resourceSpotinstOceanAWSExtendedResourceDefinitionRead(resourceData, meta)
+
+}
+
+func createOceanAWSExtendedResourceDefinition(resourceData *schema.ResourceData, erd *aws.ExtendedResourceDefinition, spotinstClient *Client) (*string, error) {
+	if json, err := commons.ToJson(erd); err != nil {
+		return nil, err
+	} else {
+		log.Printf("===> ExtendedResourceDefinition create configuration: %s", json)
+	}
+	var resp *aws.CreateExtendedResourceDefinitionOutput = nil
+	err := resource.Retry(time.Minute, func() *resource.RetryError {
+		input := &aws.CreateExtendedResourceDefinitionInput{ExtendedResourceDefinition: erd}
+		r, err := spotinstClient.ocean.CloudProviderAWS().CreateExtendedResourceDefinition(context.Background(), input)
+		if err != nil {
+			// Checks whether we should retry the ExtendedResourceDefinition creation.
+			if errs, ok := err.(client.Errors); ok && len(errs) > 0 {
+				for _, err := range errs {
+					if err.Code == "InvalidParameterValue" &&
+						strings.Contains(err.Message, "Invalid IAM Instance Profile") {
```

Comment:
`strings.Contains(err.Message, "Invalid IAM Instance Profile")` is not belong here(?)